pandas/__init__.py

```diff
@@ -14,7 +14,10 @@
 
 if _missing_dependencies:  # pragma: no cover
     raise ImportError(
-        "Unable to import required dependencies:\n" + "\n".join(_missing_dependencies)
+        "Unable to import required dependencies:\n"
+        + "\n".join(_missing_dependencies)
+        + "\n\nTo see the full error message, "
```

Comment:
nit: You can remove the `+ ` on each line; Python will perform string concatenation automatically.

Also, I think we should remove one of the `\n` here:

```
ImportError: Unable to import required dependencies:
dateutil: No module named 'dateutil'

To see the full error message, try importing the missing dependencies directly.
```

vs

```
ImportError: Unable to import required dependencies:
dateutil: No module named 'dateutil'
To see the full error message, try importing the missing dependencies directly.
```

By removing the blank line, it is more apparent to me that it's all part of the same error message.

Reply:
Hmm what if we just change this to

```python
for _dependency in _hard_dependencies:
    try:
        __import__(_dependency)
    except ImportError as _e:  # pragma: no cover
       raise ImportError(f"Unable to import required dependency {_dependency} ...") from _e
```

I know we won't be able to collect all the missing dependencies all at once but at least we'll maintain the traceback of the failed import

Reply:
@mroeschke - I'm good with this approach.